src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowHistorySpec.groovy

```diff
@@ -0,0 +1,125 @@
+    }
+    @Ignore("https://github.com/telstra/open-kilda/issues/5320")
+    @Tidy
+    def "User can change a ha-flow and get its history event - #description"() {
+        given: "Existing ha-flow"
+        def swT = topologyHelper.switchTriplets[0]
+        def haFlowRequest = haFlowHelper.randomHaFlow(swT)
+        def haFlow = haFlowHelper.addHaFlow(haFlowRequest)
+
+        when: "#description the flow"
+        change(haFlow)
+
+        then: "Ha flow update event appears"
+        expectedAssertion(haFlow)
+
+        cleanup:
+        haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+
+        where:
+        description | change | expectedAssertion
+
+        "update"    |
+                { HaFlow flow ->
+                    def allowedSharedPorts = topology.getAllowedPortsForSwitch(topology.find(
+                            flow.sharedEndpoint.switchId)) - flow.sharedEndpoint.portNumber
+                    flow.sharedEndpoint.portNumber = allowedSharedPorts[0]
+                    def updatedHaFlowPayload = haFlowHelper.convertToUpdate(flow)
+                    haFlowHelper.updateHaFlow(flow.haFlowId, updatedHaFlowPayload)
+
+                }            | { HaFlow flow -> haFlowHelper.getHistory(flow.haFlowId).getUpdateEntries().size() == 1}
+        "delete"    |
+                { HaFlow flow ->
+                    haFlowHelper.deleteHaFlow(flow.haFlowId)
+
+                }            | { HaFlow flow -> haFlowHelper.getHistory(flow.haFlowId).getDeleteEntries().size() == 1}
+
+        "reroute"    |
+                { HaFlow flow ->
+                    haFlowHelper.rerouteHaFlow(flow.haFlowId)
+
+                }            | { HaFlow flow -> haFlowHelper.getHistory(flow.haFlowId).getRerouteEntries().size() == 1}
+    }
+
+
+    def "History records can be get with timestamp filters"() {
+        given: "Ha flow"
+        def timestampBeforeCreate = System.currentTimeSeconds()
+        def swT = topologyHelper.getAllNotNeighbouringSwitchTriplets().shuffled().first()
+        def haFlow = haFlowHelper.addHaFlow(haFlowHelper.randomHaFlow(swT))
+
+        when: "Delete HA flow"
+        def timestampBeforeDelete = System.currentTimeSeconds()
+        haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+        Wrappers.wait(WAIT_OFFSET) { assert !northboundV2.getHaFlow(haFlow.haFlowId) }
+        def flowRemoved = true
+
+        then: "Possible to get ha flow history events with timestamp filters"
+        def timestampAfterDelete = System.currentTimeSeconds()
+        haFlowHelper.getHistory(haFlow.haFlowId, timestampBeforeCreate, timestampAfterDelete).entries.size() == 2
+        haFlowHelper.getHistory(haFlow.haFlowId, timestampBeforeDelete, timestampAfterDelete).getDeleteEntries().size() == 1
+
+        cleanup:
+        haFlow && !flowRemoved && haFlowHelper.deleteHaFlow(haFlow.haFlowId)
+
+    }
+
+    def "Empty history returned in case filters return no results (dateTo < dateBefore)"() {
```

Comment:
Actually, you have to receive validation error from northbound when you try to send a request with controversial parameters.
Just query time interval where nothing happened to Ha-Flow, e.g. after it was deleted.

Reply:
fixed (and agree with you)
but we have no validation on it, its how it was implemented and checked for regular flow

Reply:
So, there should be opened bug, I think.

Reply:
https://github.com/telstra/open-kilda/issues/5330